pkg/credentials/iam_aws.go

```diff
@@ -181,10 +181,6 @@ type ec2RoleCredRespBody struct {
 // be sent to fetch the rolling access credentials.
 // http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
 func getIAMRoleURL(endpoint string) (*url.URL, error) {
-	if endpoint == "" {
-		endpoint = defaultIAMRoleEndpoint
-	}
-
```

Comment:
why is this removed?

Reply:
It seemed most appropriate to hoist the block up to `getCredentials()`. I can keep it, but it seems like dead code given it appears that only `getCredentials()` calls `getIAMRoleURL()`.

Reply:
okay